core/src/main/java/org/bitcoinj/crypto/DeterministicKey.java

```diff
@@ -42,6 +42,8 @@
 
     private final DeterministicKey parent;
     private final ImmutableList<ChildNumber> childNumberPath;
+    private final int depth;
+    private final byte[] parentFingerprint; // 4 bytes if known, otherwise empty
```

Comment:
It's probably nicer if this is @Nullable as well and then we don't waste memory on byte[0] allocations that have no purpose.